src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -812,6 +831,38 @@ private static bool CanReplace(ISymbol symbol)
             return memberAccess.CanReplaceWithReducedName(replacementNode, semanticModel, cancellationToken);
         }
 
+        /// <summary>
+        /// Determines if <paramref name="speculativeSymbol"/> contains a superset of the symbols in
+        /// <paramref name="actualSymbol"/>.
+        /// </summary>
+        private static bool IsReplacementCandidate(SymbolInfo actualSymbol, SymbolInfo speculativeSymbol)
+        {
+            if (!(speculativeSymbol.Symbol is null))
+            {
+                return Equals(actualSymbol.Symbol, speculativeSymbol.Symbol);
+            }
```

Comment:
:memo: This code was rewritten after finding bugs in the original implementation.